src/libs/actions/Report.ts

```diff
@@ -4588,6 +4599,262 @@ function clearDeleteTransactionNavigateBackUrl() {
+                },
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`,
+                    value: null,
+                },
+            );
+
+            failureData.push(
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.TRANSACTION}${transactionID}`,
+                    value: {reportID: transaction?.reportID},
+                },
+                {
+                    onyxMethod: Onyx.METHOD.MERGE,
+                    key: `${ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS}${transactionID}`,
+                    value: transactionViolations,
+                },
+            );
+        }
+
+        // 2. Move the report action to self DM
+        const updatedReportAction = {
+            ...reportAction,
+            originalMessage: {
+                // eslint-disable-next-line deprecation/deprecation
+                ...reportAction.originalMessage,
+                IOUReportID: CONST.REPORT.UNREPORTED_REPORT_ID,
+                type: CONST.IOU.TYPE.TRACK,
+            },
+            reportActionID: newReportActionID,
+            pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        };
+
+        optimisticData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${selfDMReportID}`,
+            value: {[newReportActionID]: updatedReportAction},
+        });
+
+        successData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${selfDMReportID}`,
+            value: {[newReportActionID]: {pendingAction: null}},
+        });
+
+        failureData.push({
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${selfDMReportID}`,
+            value: {[newReportActionID]: null},
+        });
+
+        // 3. Update transaction thread
+        optimisticData.push(
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT}${childReportID}`,
+                value: {
+                    parentReportActionID: newReportActionID,
+                    parentReportID: selfDMReportID,
+                    chatReportID: selfDMReportID,
+                    policyID: CONST.POLICY.ID_FAKE,
+                },
+            },
+            {
+                onyxMethod: Onyx.METHOD.MERGE,
+                key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${childReportID}`,
+                value: {
+                    [newReportActionID]: {
+                        originalMessage: {
+                            movedToReportID: selfDMReportID,
+                        },
+                    },
+                },
+            },
+        );
+
+        // 4. Add UNREPORTEDTRANSACTION report action
```

Comment:
We should avoid the deleted report action which caused an issue https://github.com/Expensify/App/issues/68628